### ISSUE TITLE
mamba install throws an error for channels pointed with UNC paths

### ISSUE THREAD

**Issue by abourrier** [mamba-org/mamba#799]
Platform: Windows 10
Mamba version: 0.9

I manage several conda installations on a local network with conda channel mirrors hosted under UNC paths.
`conda install` commands work, e.g.:
`conda install pandas -c file://store01/channels/conda-forge --override-channels`.

However, the same command with `mamba install` throws the following error:
```
Traceback (most recent call last):
    File "C:\Users\Anthony\mambaforge\lib\site-packages\conda\exceptions.py", line 1079, in __call__
    return func(*args, **kwargs)
    File "C:\Users\Anthony\mambaforge\lib\site-packages\mamba\mamba.py", line 897, in exception_converter
    raise e
    File "C:\Users\Anthony\mambaforge\lib\site-packages\mamba\mamba.py", line 891, in exception_converter
    exit_code = _wrapped_main(*args, **kwargs)
    File "C:\Users\Anthony\mambaforge\lib\site-packages\mamba\mamba.py", line 850, in _wrapped_main
    result = do_call(args, p)
    File "C:\Users\Anthony\mambaforge\lib\site-packages\mamba\mamba.py", line 731, in do_call
    exit_code = install(args, parser, "install")
    File "C:\Users\Anthony\mambaforge\lib\site-packages\mamba\mamba.py", line 587, in install
    downloaded = transaction.prompt(
RuntimeError: Could not set URL (code: 3 - url = file://store01/channels/conda-forge/win-64)
```

Note that there has been an improvement with mamba 0.9 compared to mamba 0.8.2 which failed earlier in the execution.
Stack trace (mamba 0.8.2):
```
Traceback (most recent call last):
    File "C:\Users\Anthony\mambaforge\envs\mamba_test\lib\site-packages\conda\exceptions.py", line 1079, in __call__
    return func(*args, **kwargs)
    File "C:\Users\Anthony\mambaforge\envs\mamba_test\lib\site-packages\mamba\mamba.py", line 882, in exception_converter
    raise e
    File "C:\Users\Anthony\mambaforge\envs\mamba_test\lib\site-packages\mamba\mamba.py", line 876, in exception_converter
    exit_code = _wrapped_main(*args, **kwargs)
    File "C:\Users\Anthony\mambaforge\envs\mamba_test\lib\site-packages\mamba\mamba.py", line 835, in _wrapped_main
    result = do_call(args, p)
    File "C:\Users\Anthony\mambaforge\envs\mamba_test\lib\site-packages\mamba\mamba.py", line 716, in do_call
    exit_code = install(args, parser, "install")
    File "C:\Users\Anthony\mambaforge\envs\mamba_test\lib\site-packages\mamba\mamba.py", line 514, in install
    index = load_channels(pool, channels, repos)
    File "C:\Users\Anthony\mambaforge\envs\mamba_test\lib\site-packages\mamba\utils.py", line 93, in load_channels
    index = get_index(
    File "C:\Users\Anthony\mambaforge\envs\mamba_test\lib\site-packages\mamba\utils.py", line 74, in get_index
    is_downloaded = dlist.download(True)
RuntimeError: Download error (3) URL using bad/illegal format or missing URL [file://store01/channels/conda-forge/noarch/repodata.json]
```

**Comment by wolfv** [comment]
PR's welcome :) #749 was trying to fix this. 

I have no UNC url with which I could test this.

**Comment by wolfv** [comment]
@adament any ideas?

**Comment by adament** [comment]
Yes I only implemented the workaround for UNC urls in DownloadTarget, which is only a partial fix. An easy fix is simply to pass every url given to URLHandler through the same unc_url transformation. Then the question remains whether you would prefer DownloadTarget to simply use URLHandler or keep the two objects separate and just share the utility function? Which I think would then more naturally live in the URLHandler namespace.

**Comment by adament** [comment]
The problem with the easy fix would be that URLHandler(url).url(false) would not be the identity function since it would now result in a proper UNC url, and I am not certain enough about where URLHandler is used to know if that would result in issues. One could make the url function on URLHandler call fix_win_path to do the inverse transformation of unc_url which should be able to mimic current behaviour.

**Comment by adament** [comment]
I think in this instance the problem occurs because `transaction.prompt` calls `MTransaction.print` which calls `cut_repo_name` which calls `split_scheme_auth_token` which calls the constructor of `URLHandler` which passes the incorrect UNC url to curl which fails. So in this case it seems to me that either `cut_repo_name`, `split_scheme_auth_token` or `URLHandler.url` should remove the extra slashes so that the user interface is consistent(with conda as well) in not showing them - even though I think the fundamental design mistake is in stripping them in the first place.

@wolfv Do you have any preference?

**Comment by adament** [comment]
This should be fixed in #806. I fixed it by only transforming the URL given to curl, and thus the m_url variable stored on URLhandler still corresponds to the argument passed to the constructor.
